### ISSUE TITLE
Unable to slice a DBSeqRecord

### ISSUE THREAD

**Issue by ctSkennerton** [biopython/biopython#797]
I came across the following bug when trying to slice a `DBSeqRecord`

```
  File "/Users/connor/anaconda/lib/python2.7/site-packages/Bio/SeqRecord.py", line 432, in __getitem__
    description=self.description)
TypeError: __init__() got an unexpected keyword argument 'id'
```

Having a look at the relevant line (below) you can see that it is trying to construct and new record, however the `__init__` signature is different between [`SeqRecord`](https://github.com/biopython/biopython/blob/master/Bio/SeqRecord.py#L155) and [`DBSeqRecord`](https://github.com/biopython/biopython/blob/master/BioSQL/BioSeq.py#L508), which is causing the TypeError.
```
answer = self.__class__(self.seq[index],
                                    id=self.id,
                                    name=self.name,
                                    description=self.description)
```

I'd like your opinion on how to resolve this and I'll be happy to change the code and open a PR. My thought was to enforce that the new record is a `SeqRecord` rather than using `self.__class__` to dynamically select which record class since this new extracted subsequence wouldn't be found in the database. But maybe you think another way would be better?

**Comment by peterjc** [comment]
The ``DBSeq`` class should be able to slice itself and still avoid pulling the full sequence from the database until needed - and I believe that has unit tests.

It looks like an oversight that ``DBSeqRecord`` never got its own ``_getitem__`` implementation, but that seems the way forward.

It should follow the same logic used in the ``SeqRecord`` about what annotations etc should be preserved (many will be dropped), but this might be best handled as a ``SeqRecord`` using a ``DBSeq`` as the ``.seq``? Hmm. This needs some experimentation...

**Comment by ctSkennerton** [comment]
I think keeping as much of the code as possible in `SeqFeature` is probably more manageable in the long run as there would be less duplicated code. I'm not particularly sure on how to do that but we could change the signature of the `__init__` to allow `kwargs` which would alleviate the current error and still allow a `SeqRecord` or `DBSeqRecord` to be generated.

**Comment by peterjc** [comment]
I don't think changing the signature of the ``DBSeqRecord`` ``__init__`` will help - it is not designed to accept an identifier/name/description etc.

Regarding the features, currently the ``DBSeqRecord`` is quite simple: Initially it never looks at the feature tables, but will load all the features for the record via ``_retrieve_features`` if the features property is accessed, creating a list of standalone ``SeqFeature`` objects in memory - exactly like a real ``SeqRecord``.

I think the fix here is in ``SeqRecord`` method ``_getitem__`` replace:

```
            answer = self.__class__(self.seq[index],
                                    id=self.id,
                                    name=self.name,
                                    description=self.description)
```

with:

```
            answer = SeqRecord(self.seq[index],
                                    id=self.id,
                                    name=self.name,
                                    description=self.description)
```

when ``isinstance(self, DBSeqRecord)`` to avoid breaking any user-defined subclasses which are working?

**Comment by peterjc** [comment]
Fixed via #803
